akka-cluster-typed/src/main/scala/akka/cluster/typed/internal/receptionist/Registry.scala

```diff
@@ -23,23 +23,32 @@ import scala.concurrent.duration.Deadline
       val key = ORMultiMapKey[ServiceKey[_], Entry](s"ReceptionistKey_$n")
       key -> new ServiceRegistry(EmptyORMultiMap)
     }.toMap
-    new ShardedServiceRegistry(emptyRegistries, Map.empty)
+    new ShardedServiceRegistry(emptyRegistries, Map.empty, Set.empty, Map.empty)
   }
 
 }
 
 /**
+ * INTERNAL API
+ *
  * Two level structure for keeping service registry to be able to shard entries over multiple ddata keys (to not
  * get too large ddata messages)
  *
  * @param tombstones Local actors that were stopped and should not be re-added to the available set of actors
  *                   for a key. Since the only way to unregister is to stop, we don't need to keep track of
  *                   the service key
- * INTERNAL API
+ *
+ * @param registeredServices Local actors that were registered and not terminated. Used to find entries
+ *                           that have been removed from ddata but are know to be existing and should be
+ *                           re-added. Can happen when talking to a node outside the cluster such as
+ *                           a restarted node with same hostname and port.
```

Comment:
the `registeredServices` is the part that I probably will remove again, but it can be useful for logging when testing this